src/task.c

```diff
@@ -748,14 +802,28 @@ task_system_create(platform_heap_id    hid,
    // task initialization
    register_init_tid_hook();
 
+   // Treat this as a soft-error and configure the task system w/o
+   // background threads when one of these configs is 0.
+   bool use_bg_threads = ((num_bg_threads[TASK_TYPE_MEMTABLE] != 0)
```

Comment:
I would prefer this to be a hard error.

Reply:
If you feel strongly about it, I'm ok changing this to a hard-error. And will rework tests accordingly.

@ajhconway -- what's your vote on this point? Should this be a soft-error (degrading to no-bg-threads enabled), or a hard-error (splinter does not come up) ?

Reply:
Vote is to make a hard error and check in the newly added validator fn.

Reply:
Done, this is now a hard-error.